diff --git a/node_modules/ava/lib/load-config.js b/node_modules/ava/lib/load-config.js
index e3c70fd..9e2d490 100644
--- a/node_modules/ava/lib/load-config.js
+++ b/node_modules/ava/lib/load-config.js
@@ -38,7 +38,7 @@ function resolveConfigFile(configFile) {
 		configFile = path.resolve(configFile); // Relative to CWD
 
 		if (!configFile.endsWith('.js') && !configFile.endsWith('.cjs') && !configFile.endsWith('.mjs')) {
-			throw new Error('Config files must have .js, .cjs or .mjs extensions');
+			// throw new Error('Config files must have .js, .cjs or .mjs extensions');
 		}
 	}
 
