diff --git a/node_modules/typedoc-plugin-markdown/dist/resources/helpers/toc.js b/node_modules/typedoc-plugin-markdown/dist/resources/helpers/toc.js
index e332374..1ecd370 100644
--- a/node_modules/typedoc-plugin-markdown/dist/resources/helpers/toc.js
+++ b/node_modules/typedoc-plugin-markdown/dist/resources/helpers/toc.js
@@ -25,6 +25,15 @@ var __importStar = (this && this.__importStar) || function (mod) {
 Object.defineProperty(exports, "__esModule", { value: true });
 const Handlebars = __importStar(require("handlebars"));
 const utils_1 = require("../../utils");
+function reorderHead(input, times) {
+    input.push(input.shift());
+    times--;
+    if (times !== 0) {
+        return reorderHead(input, times);
+    }
+    return input;
+}
+const reorderCount = 3
 function default_1(theme) {
     Handlebars.registerHelper('toc', function () {
         var _a, _b;
@@ -40,7 +49,12 @@ function default_1(theme) {
                 md.push(`## Table of contents\n\n`);
             }
             const headingLevel = hideInPageTOC ? `##` : `###`;
-            (_b = this.groups) === null || _b === void 0 ? void 0 : _b.forEach((group) => {
+            (_b = this.groups) === null || _b === void 0 ? void 0 : reorderHead(_b, reorderCount).forEach((group, index, array) => {
+
+                if (this.url === 'README.md' && index === array.length - reorderCount) {
+                    md.push(`<details><summary>Uncategorized</summary>\n\n`)
+                }
+
                 const groupTitle = group.title;
                 if (group.categories) {
                     group.categories.forEach((category) => {
@@ -56,6 +70,10 @@ function default_1(theme) {
                         md.push('\n');
                     }
                 }
+
+                if (this.url === 'README.md' && index === array.length - 1) {
+                    md.push(`</details>`)
+                }
             });
         }
         return md.length > 0 ? md.join('\n') : null;
diff --git a/node_modules/typedoc-plugin-markdown/dist/resources/partials/member.declaration.hbs b/node_modules/typedoc-plugin-markdown/dist/resources/partials/member.declaration.hbs
index 1ca0b74..ac61b79 100755
--- a/node_modules/typedoc-plugin-markdown/dist/resources/partials/member.declaration.hbs
+++ b/node_modules/typedoc-plugin-markdown/dist/resources/partials/member.declaration.hbs
@@ -14,58 +14,4 @@
 
 {{/if}}
 
-{{#if type.declaration}}
-
-{{#if type.declaration.indexSignature}}
-
-{{#with type.declaration.indexSignature}}
-
-#### Index signature
-
-{{{indexSignatureTitle}}}
-
-{{> comment}}
-
-{{/with}}
-
-{{/if}}
-
-{{#if type.declaration.signatures}}
-
-{{#if type.declaration.children}}
-
-#### Call signature
-
-{{else}}
-
-#### Type declaration
-
-{{/if}}
-
-{{#each type.declaration.signatures}}
-
-{{> member.signature showSources=false }}
-
-{{/each}}
-
-{{/if}}
-
-{{#if type.declaration.children}}
-
-{{#with type.declaration}}
-
-#### Type declaration
-
-{{/with}}
-
-{{#with type.declaration.children}}
-
-{{{typeDeclarationMembers}}}
-
-{{/with}}
-
-{{/if}}
-
-{{/if}}
-
 {{> member.sources}}
\ No newline at end of file
diff --git a/node_modules/typedoc-plugin-markdown/dist/resources/partials/member.sources.hbs b/node_modules/typedoc-plugin-markdown/dist/resources/partials/member.sources.hbs
index a999ee8..3f11cfe 100755
--- a/node_modules/typedoc-plugin-markdown/dist/resources/partials/member.sources.hbs
+++ b/node_modules/typedoc-plugin-markdown/dist/resources/partials/member.sources.hbs
@@ -10,30 +10,6 @@
 
 {{/if}}
 
-{{#if inheritedFrom}}
-
-#### Inherited from
-
-{{#with inheritedFrom}}
-
-{{{typeAndParent}}}
-
-{{/with}}
-
-{{/if}}
-
-{{#if overwrites}}
-
-#### Overrides
-
-{{#with overwrites}}
-
-{{typeAndParent}}
-
-{{/with}}
-
-{{/if}}
-
 {{#if sources}}
 
 #### Defined in
diff --git a/node_modules/typedoc-plugin-markdown/dist/resources/partials/title.hbs b/node_modules/typedoc-plugin-markdown/dist/resources/partials/title.hbs
index ba8e5be..6a4e1d3 100644
--- a/node_modules/typedoc-plugin-markdown/dist/resources/partials/title.hbs
+++ b/node_modules/typedoc-plugin-markdown/dist/resources/partials/title.hbs
@@ -4,4 +4,4 @@
 
 {{/ifShowPageTitle}}
 
-{{{reflectionPath}}}
\ No newline at end of file
+[💗 Help the project](https://github.com/sponsors/panva)
diff --git a/node_modules/typedoc-plugin-markdown/dist/resources/templates/reflection.hbs b/node_modules/typedoc-plugin-markdown/dist/resources/templates/reflection.hbs
index ce29bc5..6833f14 100755
--- a/node_modules/typedoc-plugin-markdown/dist/resources/templates/reflection.hbs
+++ b/node_modules/typedoc-plugin-markdown/dist/resources/templates/reflection.hbs
@@ -24,18 +24,6 @@
 
 {{/if}}
 
-{{#ifShowTypeHierarchy}}
-
-## Hierarchy
-
-{{#with model.typeHierarchy}}
-
-{{{hierarchy 0}}}
-
-{{/with}}
-
-{{/ifShowTypeHierarchy}}
-
 {{#if model.implementedTypes}}
 
 ## Implements
@@ -97,3 +85,15 @@
 {{> main}}
 
 {{/with}}
+
+{{#ifShowTypeHierarchy}}
+
+## Hierarchy
+
+{{#with model.typeHierarchy}}
+
+{{{hierarchy 0}}}
+
+{{/with}}
+
+{{/ifShowTypeHierarchy}}
\ No newline at end of file
diff --git a/node_modules/typedoc-plugin-markdown/dist/theme.js b/node_modules/typedoc-plugin-markdown/dist/theme.js
index 2f6715b..de5cf7f 100644
--- a/node_modules/typedoc-plugin-markdown/dist/theme.js
+++ b/node_modules/typedoc-plugin-markdown/dist/theme.js
@@ -145,7 +145,7 @@ class MarkdownTheme extends typedoc_1.Theme {
         });
     }
     toAnchorRef(reflectionId) {
-        return reflectionId;
+        return reflectionId.startsWith('[') && reflectionId.endsWith(']') ? reflectionId.slice(1, -1) : reflectionId;
     }
     getRelativeUrl(absolute) {
         if (MarkdownTheme.URL_PREFIX.test(absolute)) {
