diff --git a/node_modules/typedoc/dist/lib/converter/plugins/CommentPlugin.js b/node_modules/typedoc/dist/lib/converter/plugins/CommentPlugin.js
index 3937272..0fdc70d 100644
--- a/node_modules/typedoc/dist/lib/converter/plugins/CommentPlugin.js
+++ b/node_modules/typedoc/dist/lib/converter/plugins/CommentPlugin.js
@@ -373,9 +373,9 @@ function mergeSeeTags(comment) {
     const index = comment.blockTags.indexOf(see[0]);
     comment.removeTags("@see");
     see[0].content = see.flatMap((part) => [
+        { kind: "text", text: "\n" },
         { kind: "text", text: " - " },
         ...part.content,
-        { kind: "text", text: "\n" },
     ]);
     comment.blockTags.splice(index, 0, see[0]);
 }
