diff --git a/examples/oauth.ts b/examples/client_secret_basic.ts
index d55e62d..e1487c0 100644
--- a/examples/oauth.ts
+++ b/examples/client_secret_basic.ts
@@ -23,7 +23,7 @@ const as = await oauth
   .then((response) => oauth.processDiscoveryResponse(issuer, response))
 
 const client: oauth.Client = { client_id }
-const clientAuth = oauth.ClientSecretPost(client_secret)
+const clientAuth = oauth.ClientSecretBasic(client_secret)
 
 const code_challenge_method = 'S256'
 /**
