diff --git a/examples/oauth.ts b/examples/client_secret_post.ts
index a3b923c..6185acc 100644
--- a/examples/oauth.ts
+++ b/examples/client_secret_post.ts
@@ -24,7 +24,7 @@ const as = await oauth
 const client: oauth.Client = {
   client_id,
   client_secret,
-  token_endpoint_auth_method: 'client_secret_basic',
+  token_endpoint_auth_method: 'client_secret_post',
 }
 
 const code_challenge_method = 'S256'
