diff --git a/examples/code.ts b/examples/dpop.ts
index f5f3315..02e8357 100644
--- a/examples/code.ts
+++ b/examples/dpop.ts
@@ -1,5 +1,9 @@
 import * as oauth from '../src/index.js'
 
+// in order to take full advantage of DPoP you shall generate a random, non-extractable private key for every session
+// in the browser environment you shall use IndexedDB to persist the generated CryptoKeyPair
+const DPoP = await oauth.generateKeyPair('ES256')
+
 const issuer = new URL('https://op.panva.cz')
 const as = await oauth
   .discoveryRequest(issuer)
@@ -56,12 +60,16 @@ let access_token: string
     params,
     redirect_uri,
     code_verifier,
+    { DPoP },
   )
 
   let challenges: oauth.WWWAuthenticateChallenge[] | undefined
   if ((challenges = oauth.parseWwwAuthenticateChallenges(response))) {
     for (const challenge of challenges) {
       console.log('challenge', challenge)
+      if (challenge.scheme === 'dpop' && challenge.parameters.error === 'use_dpop_nonce') {
+        // the AS-signalled nonce is now already cached, you should retry `authorizationCodeGrantRequest`
+      }
     }
     throw new Error() // Handle www-authenticate challenges as needed
   }
@@ -81,12 +89,15 @@ let access_token: string
 
 // fetch userinfo response
 {
-  const response = await oauth.userInfoRequest(as, client, access_token)
+  const response = await oauth.userInfoRequest(as, client, access_token, { DPoP })
 
   let challenges: oauth.WWWAuthenticateChallenge[] | undefined
   if ((challenges = oauth.parseWwwAuthenticateChallenges(response))) {
     for (const challenge of challenges) {
       console.log('challenge', challenge)
+      if (challenge.scheme === 'dpop' && challenge.parameters.error === 'use_dpop_nonce') {
+        // the AS-signalled nonce is now already cached, you should retry `userInfoRequest`
+      }
     }
     throw new Error() // Handle www-authenticate challenges as needed
   }
