diff --git a/examples/fapi2.ts b/examples/fapi2-message-signing.ts
index 80ec0f4..c4b4e4f 100644
--- a/examples/fapi2.ts
+++ b/examples/fapi2-message-signing.ts
@@ -24,6 +24,11 @@ let DPoP!: CryptoKeyPair
  * client authentication method.
  */
 let clientPrivateKey!: CryptoKey
+/**
+ * A key that is pre-registered at the Authorization Server that the client is supposed to sign its
+ * Request Objects with.
+ */
+let jarPrivateKey!: CryptoKey
 
 // End of prerequisites
 
@@ -45,8 +50,8 @@ const code_challenge_method = 'S256'
 const code_verifier = oauth.generateRandomCodeVerifier()
 const code_challenge = await oauth.calculatePKCECodeChallenge(code_verifier)
 
-// Pushed Authorization Request & Response (PAR)
-let request_uri: string
+// Signed Request Object (JAR)
+let request: string
 {
   const params = new URLSearchParams()
   params.set('client_id', client.client_id)
@@ -55,6 +60,17 @@ let request_uri: string
   params.set('redirect_uri', redirect_uri)
   params.set('response_type', 'code')
   params.set('scope', 'api:read')
+  params.set('response_mode', 'jwt')
+
+  request = await oauth.issueRequestObject(as, client, params, jarPrivateKey)
+}
+
+// Pushed Authorization Request & Response (PAR)
+let request_uri: string
+{
+  const params = new URLSearchParams()
+  params.set('client_id', client.client_id)
+  params.set('request', request)
 
   const pushedAuthorizationRequest = () =>
     oauth.pushedAuthorizationRequest(as, client, params, {
@@ -105,7 +121,7 @@ let access_token: string
 {
   // @ts-expect-error
   const currentUrl: URL = getCurrentUrl()
-  const params = oauth.validateAuthResponse(as, client, currentUrl)
+  const params = await oauth.validateJwtAuthResponse(as, client, currentUrl)
   if (oauth.isOAuth2Error(params)) {
     console.error('Error Response', params)
     throw new Error() // Handle OAuth 2.0 redirect error
@@ -142,6 +158,9 @@ let access_token: string
     }
   }
 
+  // Check ID Token signature for non-repudiation purposes
+  await oauth.validateIdTokenSignature(as, result)
+
   console.log('Access Token Response', result)
   ;({ access_token } = result)
 }
