diff --git a/examples/fapi2.ts b/examples/fapi2-message-signing.ts
index 2f73b7a..f49f057 100644
--- a/examples/fapi2.ts
+++ b/examples/fapi2-message-signing.ts
@@ -20,6 +20,11 @@ let DPoP!: CryptoKeyPair
  * client authentication method.
  */
 let clientPrivateKey!: CryptoKey
+/**
+ * A key that is pre-registered at the Authorization Server that the client is supposed to sign its
+ * Request Objects with.
+ */
+let jarPrivateKey!: CryptoKey
 
 // End of prerequisites
 
@@ -36,7 +41,7 @@ const code_verifier = oauth.generateRandomCodeVerifier()
 const code_challenge = await oauth.calculatePKCECodeChallenge(code_verifier)
 const code_challenge_method = 'S256'
 
-let request_uri: string
+let request: string
 {
   const params = new URLSearchParams()
   params.set('client_id', client.client_id)
@@ -44,8 +49,18 @@ let request_uri: string
   params.set('code_challenge_method', code_challenge_method)
   params.set('redirect_uri', redirect_uri)
   params.set('response_type', 'code')
+  params.set('response_mode', 'jwt')
   params.set('scope', 'openid email')
 
+  request = await oauth.issueRequestObject(as, client, params, jarPrivateKey)
+}
+
+let request_uri: string
+{
+  const params = new URLSearchParams()
+  params.set('client_id', client.client_id)
+  params.set('request', request)
+
   const response = await oauth.pushedAuthorizationRequest(as, client, params, {
     DPoP,
     clientPrivateKey,
@@ -82,7 +97,7 @@ let request_uri: string
 {
   // @ts-expect-error
   const currentUrl: URL = getCurrentUrl()
-  const params = oauth.validateAuthResponse(as, client, currentUrl, oauth.expectNoState)
+  const params = await oauth.validateJwtAuthResponse(as, client, currentUrl, oauth.expectNoState)
   if (oauth.isOAuth2Error(params)) {
     console.log('error', params)
     throw new Error() // Handle OAuth 2.0 redirect error
