diff --git a/examples/code.ts b/examples/jarm.ts
index 66a88e8..ad8fa61 100644
--- a/examples/code.ts
+++ b/examples/jarm.ts
@@ -30,6 +30,7 @@ const code_challenge_method = 'S256'
   authorizationUrl.searchParams.set('redirect_uri', redirect_uri)
   authorizationUrl.searchParams.set('response_type', 'code')
   authorizationUrl.searchParams.set('scope', 'openid email')
+  authorizationUrl.searchParams.set('response_mode', 'jwt')
 }
 
 // one eternity later, the user lands back on the redirect_uri
@@ -38,7 +39,7 @@ let access_token: string
 {
   // @ts-ignore
   const currentUrl: URL = getCurrentUrl()
-  const parameters = lib.validateAuthResponse(as, client, currentUrl, lib.expectNoState)
+  const parameters = await lib.validateJwtAuthResponse(as, client, currentUrl, lib.expectNoState)
   if (lib.isOAuth2Error(parameters)) {
     console.log('error', parameters)
     throw new Error() // Handle OAuth 2.0 redirect error
