diff --git a/examples/code.ts b/examples/jarm.ts
index 17d6b71..f04c0d4 100644
--- a/examples/code.ts
+++ b/examples/jarm.ts
@@ -34,6 +34,7 @@ const code_challenge_method = 'S256'
   authorizationUrl.searchParams.set('redirect_uri', redirect_uri)
   authorizationUrl.searchParams.set('response_type', 'code')
   authorizationUrl.searchParams.set('scope', 'openid email')
+  authorizationUrl.searchParams.set('response_mode', 'jwt')
 }
 
 // one eternity later, the user lands back on the redirect_uri
@@ -42,7 +43,7 @@ let access_token: string
 {
   // @ts-ignore
   const currentUrl: URL = getCurrentUrl()
-  const params = oauth.validateAuthResponse(as, client, currentUrl, oauth.expectNoState)
+  const params = await oauth.validateJwtAuthResponse(as, client, currentUrl, oauth.expectNoState)
   if (oauth.isOAuth2Error(params)) {
     console.log('error', params)
     throw new Error() // Handle OAuth 2.0 redirect error
