diff --git a/examples/code.ts b/examples/private_key_jwt.ts
index a7303f1..9a34637 100644
--- a/examples/code.ts
+++ b/examples/private_key_jwt.ts
@@ -4,12 +4,16 @@ import * as oauth from '../src/index.js' // replace with an import of oauth4weba
 
 let issuer!: URL // Authorization server's Issuer Identifier URL
 let client_id!: string
-let client_secret!: string
 /**
  * Value used in the authorization request as redirect_uri pre-registered at the Authorization
  * Server.
  */
 let redirect_uri!: string
+/**
+ * A key that the client has pre-registered at the Authorization Server for use with Private Key JWT
+ * client authentication method.
+ */
+let clientPrivateKey!: CryptoKey
 
 // End of prerequisites
 
@@ -19,8 +23,7 @@ const as = await oauth
 
 const client: oauth.Client = {
   client_id,
-  client_secret,
-  token_endpoint_auth_method: 'client_secret_basic',
+  token_endpoint_auth_method: 'private_key_jwt',
 }
 
 const code_challenge_method = 'S256'
@@ -73,6 +76,7 @@ let access_token: string
     params,
     redirect_uri,
     code_verifier,
+    { clientPrivateKey },
   )
 
   let challenges: oauth.WWWAuthenticateChallenge[] | undefined
