diff --git a/examples/oauth.ts b/examples/private_key_jwt.ts
index 6f4b65e..d13c813 100644
--- a/examples/oauth.ts
+++ b/examples/private_key_jwt.ts
@@ -9,12 +9,16 @@ let algorithm!:
   | 'oidc' /* For .well-known/openid-configuration discovery */
   | undefined /* Defaults to 'oidc' */
 let client_id!: string
-let client_secret!: string
 /**
  * Value used in the authorization request as redirect_uri pre-registered at the Authorization
  * Server.
  */
 let redirect_uri!: string
+/**
+ * A key that the client has pre-registered at the Authorization Server for use with Private Key JWT
+ * client authentication method.
+ */
+let clientPrivateKey!: oauth.CryptoKey
 
 // End of prerequisites
 
@@ -24,8 +28,7 @@ const as = await oauth
 
 const client: oauth.Client = {
   client_id,
-  client_secret,
-  token_endpoint_auth_method: 'client_secret_basic',
+  token_endpoint_auth_method: 'private_key_jwt',
 }
 
 const code_challenge_method = 'S256'
@@ -73,6 +76,7 @@ let access_token: string
     params,
     redirect_uri,
     code_verifier,
+    { clientPrivateKey },
   )
 
   const result = await oauth.processAuthorizationCodeOAuth2Response(as, client, response)
