diff --git a/examples/code.ts b/examples/public.ts
index 895d96e..ba360f4 100644
--- a/examples/code.ts
+++ b/examples/public.ts
@@ -4,7 +4,6 @@ import * as oauth from '../src/index.js' // replace with an import of oauth4weba
 
 let issuer!: URL // Authorization server's Issuer Identifier URL
 let client_id!: string
-let client_secret!: string
 /**
  * Value used in the authorization request as redirect_uri pre-registered at the Authorization
  * Server.
@@ -19,8 +18,7 @@ const as = await oauth
 
 const client: oauth.Client = {
   client_id,
-  client_secret,
-  token_endpoint_auth_method: 'client_secret_basic',
+  token_endpoint_auth_method: 'none',
 }
 
 if (as.code_challenge_methods_supported?.includes('S256') !== true) {
