diff --git a/examples/oauth.ts b/examples/public.ts
index a3321ad..6a55551 100644
--- a/examples/oauth.ts
+++ b/examples/public.ts
@@ -8,7 +8,6 @@ let algorithm!:
   | 'oidc' /* For .well-known/openid-configuration discovery */
   | undefined /* Defaults to 'oidc' */
 let client_id!: string
-let client_secret!: string
 /**
  * Value used in the authorization request as redirect_uri pre-registered at the Authorization
  * Server.
@@ -23,8 +22,7 @@ const as = await oauth
 
 const client: oauth.Client = {
   client_id,
-  client_secret,
-  token_endpoint_auth_method: 'client_secret_basic',
+  token_endpoint_auth_method: 'none',
 }
 
 const code_challenge_method = 'S256'
